wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/SerializableConverter.java

```diff
@@ -47,7 +49,8 @@ public String convertToDatabaseColumn(Serializable value) {
   @Override
   public Serializable convertToEntityAttribute(String dbData) {
     try {
-      return objectMapper.readValue(dbData, Serializable.class);
+      Serializable[] arr = objectMapper.readValue(dbData, Serializable[].class);
+      return (arr.length == 1) ? arr[0] : arr;
```

Comment:
What about legit 1 element arrays?

Reply:
Good question..
Any ideas better than:
```
try {
  objectMapper.readValue(dbData, Serializable[].class);
} catch (Exc e) {  //not an array
  objectMapper.readValue(dbData, Serializable.class);
}
```
?

Reply:
What about reading it as json nodes and see if it is an array node or a value node?

Reply:
little overhead, but probably will work..